### ISSUE TITLE
segfault with jpg

### ISSUE THREAD

**Issue by mihandrei** [i3/i3lock#186]
jpg seems not to be supported but you should avoid segfaulting.

i3lock: version 2.9.1

``` bash
$ i3lock -i lala.jpg
segmentation fault (core dumped)

(gdb) backtrace
 in png_get_IHDR () at /lib64/libpng16.so.16
 in read_png () at /lib64/libcairo.so.2
 in cairo_image_surface_create_from_png ()
    at /lib64/libcairo.so.2
 in main ()

$ ldd /usr/bin/i3lock
  libcairo.so.2 => /lib64/libcairo.so.2

$ ldconfig -v | grep cairo
libcairo.so.2 -> libcairo.so.2.11510.0
  
```
Installed on fedora 27 via dnf

**Comment by mihandrei** [comment]
It seems to be cairo's fault. 
This hangs:

```python
from ctypes import *

def main(pth):
    pth = pth.encode()                                                   
    lib = cdll.LoadLibrary('/lib64/libcairo.so.2')
    cpth = c_char_p(pth)
    ret = lib.cairo_image_surface_create_from_png(cpth)    
```

**Comment by Airblader** [comment]
Thanks for investigating. I think this is problably resolved by #171 already, but the next version hasn't yet been released. Could you please confirm that the current git HEAD doesn't behave this way?

**Comment by stapelberg** [comment]
This is very likely solved with #171, which is included in i3lock 2.11. Closing.
